libgalois/include/katana/PropertyGraph.h

```diff
@@ -29,12 +29,81 @@ struct KATANA_EXPORT GraphTopology {
   using edges_range = StandardRange<edge_iterator>;
   using iterator = node_iterator;
 
+private:
+  LargeArray<Edge> adj_indices_;
+  LargeArray<Node> dests_;
+
+  // TODO(amber): make these private in the near future. No other class should
+  // access these directly. Instead provide access methods
+public:
   std::shared_ptr<arrow::UInt64Array> out_indices;
   std::shared_ptr<arrow::UInt32Array> out_dests;
 
-  uint64_t num_nodes() const { return out_indices ? out_indices->length() : 0; }
+private:
+  // TODO: generalize to typename T
+  static std::shared_ptr<arrow::UInt64Array> largeToArrowArray(
+      LargeArray<uint64_t>& lgArr) noexcept {
+    return std::make_shared<arrow::UInt64Array>(
+        lgArr.size(), arrow::MutableBuffer::Wrap(lgArr.data(), lgArr.size()));
+  }
+
+  static std::shared_ptr<arrow::UInt32Array> largeToArrowArray(
+      LargeArray<uint32_t>& lgArr) noexcept {
+    return std::make_shared<arrow::UInt32Array>(
+        lgArr.size(), arrow::MutableBuffer::Wrap(lgArr.data(), lgArr.size()));
+  }
+
+  GraphTopology(const GraphTopology&) = delete;
+  GraphTopology& operator=(const GraphTopology&) = delete;
+
+public:
+  GraphTopology() = default;
+
+  GraphTopology(
+      const Edge* adjIndices, size_t numNodes, const Node* dests,
```

Comment:
style: `snake_case` for variables

Reply:
`const Edge*` to `const Edge&` ?
Also,
`const Node*` to `const Node&`?

Reply:
The intent is to get a pointer to a buffer or array. With const ref, someone can pass a temporary and it'll compile just fine.